spring-integration-core/src/main/java/org/springframework/integration/endpoint/AbstractPollingEndpoint.java

```diff
@@ -343,13 +345,16 @@ public void run() {
 						count++;
 					}
 					catch (Exception e) {
-						if (e instanceof RuntimeException) {
-							throw (RuntimeException) e;
+						if (e instanceof MessagingException) {
+							throw (MessagingException) e;
 						}
 						else {
-							throw new MessageHandlingException(new ErrorMessage(e), e);
+							throw new MessagingException(messageHolder.get(), e);
```

Comment:
I think, I'm fine with the solution.
I thought that we don't that `ThreadLocal` when we have a TX, but indeed it is already finished and cleaned on the `Poller.this.pollingTask.call()`. And if TX throws exception during commit, we really don't have a TX-synched resource to consult for the `Message`.

So, LGTM here, although I still think that it should be `MessageHandlingException`. Just because it isn't so generic case, but really the `handleMessage(Message)` problem.